### ISSUE TITLE
ERROR: LoadError: UndefVarError: installed not defined

### ISSUE THREAD

**Issue by zsz00** [JuliaCN/Py2Jl.jl#12]
Julia Version 1.4.0-DEV.254
Commit 3b1c1fdde3 (2019-10-04 01:47 UTC)
Platform Info:
  OS: Windows (x86_64-w64-mingw32)
  CPU: Intel(R) Core(TM) i5-6200U CPU @ 2.30GHz
  WORD_SIZE: 64
  LIBM: libopenlibm
  LLVM: libLLVM-6.0.1 (ORCJIT, skylake)

using Py2Jl

ERROR: LoadError: LoadError: LoadError: UndefVarError: installed not defined
Stacktrace:
 [1] getproperty(::Module, ::Symbol) at .\Base.jl:26
 [2] top-level scope at C:\Users\zsz\.julia\packages\Py2Jl\H6uW5\src\ASTGen.jl:13
 [3] eval(::Module, ::Any) at .\boot.jl:330
....

**Comment by thautwarm** [comment]
Sorry. You should ping me...

**Comment by thautwarm** [comment]
Did `Pkg` change it API in 1.4?

**Comment by visr** [comment]
`Pkg.installed` is deprecated in 1.4, yes.

I'm not sure I understand the `compat_32842` code. Why do we check if MLStyle is installed if it is in the dependencies? Can't we just require a sufficiently new MLStyle version?

**Comment by thautwarm** [comment]
@visr You could just ask me the functionality because this is quite a tricky workaround.

It's exactly a workaround for https://github.com/JuliaLang/julia/issues/32842 .

To solve this, a quick fix is transforming all string literal pattern like `"xxx"` to pattern `&"xxx"`, where the former is matched by `===` comparisons, and the latter is matched by `==` comparisons.

**Comment by visr** [comment]
I see. Would it be an option to stop supporting the julia versions where this is broken?

**Comment by thautwarm** [comment]
@visr I don't think it an appealing idea. We can make a tag for previously supported versions. And remove `compat_32842` stuffs and poblish things since new Julia versions.

**Comment by thautwarm** [comment]
I'll do it later, but this is actually not in a high priority.

If you want I can give your help for trying maintaining it.
We could talk in Slack(search `thautwarm`/`Taine Zhao` in Julia Slack), and I can answer you any trivial or non-trivial questions about this package.

**Comment by visr** [comment]
Thanks! This issue just gives a deprecation warning on julia 1.4-rc. It was submitted during 1.4-dev, probably before the deprecation was in place, giving a hard error.
